### PR TITLE
155969 build warning for legacy code snippet

### PR DESCRIPTION
**Pull request by adunndevster**
This feature just issues a mild build warning if a content author uses the legacy code snippet.

### CODE REVIEW

**Comment by adunndevster** [comment]
@yufeih - Can you review please?

**Comment by herohua** [comment]
@adunndevster : can you also post the link to the DevOps item? Thanks!

**Comment by adunndevster** [comment]
@yufeih - Done! Good catch!

@herohua - Here it is: https://ceapex.visualstudio.com/Engineering/_boards/board/t/AXE%20Team/Stories/?workitem=155969

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs [hunk 1]
```diff
@@ -176,6 +176,8 @@ private void AddExtractorItem(string language, CodeSnippetExtrator extractor)
         protected override void Write(HtmlRenderer renderer, CodeSnippet codeSnippet)
         {
             var (content, codeSnippetPath) = _context.ReadFile(codeSnippet.CodePath, InclusionContext.File, codeSnippet);
+            
+            _context.LogWarning("code-snippet-deprecated", "The legacy code snippet inclusion will be deprecated at a later date. Please use the new code extension. (http://aka.ms/codesnippet)", codeSnippet);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Based on the comments in the linked work item, I see this change should not block completing any build currently. Some repos like `azure-docs-pr` blocks a pull request if it contains a warning. Should this be `_context.LogSuggestion`? We can change it back to `_context.LogWarning` when it is ready to be enforeced.

Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Sounds good @superyyrrzz

Comment (adunndevster) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/CodeSnippet/HtmlCodeSnippetRenderer.cs]: Done @superyyrrzz !

**Pull request merged**